cmake/nokia_ngage.cmake

```diff
@@ -0,0 +1,125 @@
+  "${SDL3_SOURCE_DIR}/src/dynapi/*.c"
+  "${SDL3_SOURCE_DIR}/src/events/*.c"
+  "${SDL3_SOURCE_DIR}/src/file/*.c"
+  "${SDL3_SOURCE_DIR}/src/file/generic/*.c"
+  "${SDL3_SOURCE_DIR}/src/filesystem/*.c"
+  "${SDL3_SOURCE_DIR}/src/filesystem/ngage/*.c"
+  "${SDL3_SOURCE_DIR}/src/filesystem/ngage/*.cpp"
+  "${SDL3_SOURCE_DIR}/src/filesystem/posix/*.c"
+  "${SDL3_SOURCE_DIR}/src/hidapi/*.c"
+  "${SDL3_SOURCE_DIR}/src/io/*.c"
+  "${SDL3_SOURCE_DIR}/src/io/generic/*.c"
+  "${SDL3_SOURCE_DIR}/src/joystick/*.c"
+  "${SDL3_SOURCE_DIR}/src/joystick/dummy/*.c"
+  "${SDL3_SOURCE_DIR}/src/libm/s_isinff.c"
+  "${SDL3_SOURCE_DIR}/src/libm/s_isnanf.c"
+  "${SDL3_SOURCE_DIR}/src/locale/*.c"
+  "${SDL3_SOURCE_DIR}/src/locale/ngage/*.cpp"
+  "${SDL3_SOURCE_DIR}/src/main/*.c"
+  "${SDL3_SOURCE_DIR}/src/main/ngage/*.c"
+  "${SDL3_SOURCE_DIR}/src/main/ngage/*.cpp"
+  "${SDL3_SOURCE_DIR}/src/misc/*.c"
+  "${SDL3_SOURCE_DIR}/src/misc/dummy/*.c"
+  "${SDL3_SOURCE_DIR}/src/power/*.c"
+  "${SDL3_SOURCE_DIR}/src/render/*.c"
+  "${SDL3_SOURCE_DIR}/src/render/ngage/*.c"
+  "${SDL3_SOURCE_DIR}/src/render/ngage/*.cpp"
+  "${SDL3_SOURCE_DIR}/src/sensor/*.c"
+  "${SDL3_SOURCE_DIR}/src/sensor/dummy/*.c"
+  "${SDL3_SOURCE_DIR}/src/stdlib/*.c"
+  "${SDL3_SOURCE_DIR}/src/storage/*.c"
+  "${SDL3_SOURCE_DIR}/src/storage/generic/*.c"
+  "${SDL3_SOURCE_DIR}/src/thread/*.c"
+  "${SDL3_SOURCE_DIR}/src/thread/generic/*.c"
+  "${SDL3_SOURCE_DIR}/src/time/*.c"
+  "${SDL3_SOURCE_DIR}/src/time/ngage/*.cpp"
+  "${SDL3_SOURCE_DIR}/src/time/unix/*.c"
+  "${SDL3_SOURCE_DIR}/src/timer/*.c"
+  "${SDL3_SOURCE_DIR}/src/timer/ngage/*.cpp"
+  "${SDL3_SOURCE_DIR}/src/tray/*.c"
+  "${SDL3_SOURCE_DIR}/src/tray/dummy/*.c"
+  "${SDL3_SOURCE_DIR}/src/video/*.c"
+  "${SDL3_SOURCE_DIR}/src/video/ngage/*.c"
+  "${SDL3_SOURCE_DIR}/src/video/yuv2rgb/*.c")
+
+add_library(${PROJECT_NAME} STATIC ${SDL3_sources})
+
+target_include_directories(
+  ${PROJECT_NAME}
+  PUBLIC
+  "${SDL3_SOURCE_DIR}/include"
+  PRIVATE
+  "${SDL3_SOURCE_DIR}/src"
+  "${SDL3_SOURCE_DIR}/include/build_config")
+
+target_compile_definitions(
+  ${PROJECT_NAME}
+  PUBLIC
+  SDL_STATIC_LIB)
+
+set(test_static_libs
+  ${CMAKE_CURRENT_BINARY_DIR}/libSDL3.a)
+
+set(test_libs
+  ${CMAKE_CURRENT_BINARY_DIR}/libSDL3.a
+  ${EPOC_LIB}/NRenderer.lib
+  ${EPOC_LIB}/3dtypes.a
+  ${EPOC_LIB}/cone.lib
+  ${EPOC_PLATFORM}/gcc/lib/gcc-lib/arm-epoc-pe/2.9-psion-98r2/libgcc.a
+  ${EPOC_PLATFORM}/ngagesdk/lib/gcc/arm-epoc-pe/4.6.4/libgcc_ngage.a
+  ${EPOC_PLATFORM}/ngagesdk/lib/gcc/arm-epoc-pe/4.6.4/libc_ngage.a
+  ${EPOC_LIB}/mediaclientaudiostream.lib
+  ${EPOC_LIB}/charconv.lib
+  ${EPOC_LIB}/bitgdi.lib
+  ${EPOC_LIB}/euser.lib
+  ${EPOC_LIB}/estlib.lib
+  ${EPOC_LIB}/ws32.lib
+  ${EPOC_LIB}/hal.lib
+  ${EPOC_LIB}/fbscli.lib
+  ${EPOC_LIB}/efsrv.lib
+  ${EPOC_LIB}/scdv.lib
+  ${EPOC_LIB}/gdi.lib)
+
+set(UID1 0x1000007a) # KExecutableImageUidValue, e32uid.h
+set(UID2 0x100039ce) # KAppUidValue16, apadef.h
+set(UID3 0x10005799) # ngage_test.exe UID
```

Comment:
I don't know an unique uid3 is required for every SDL test/example.
If so, we can perhaps create one by hashing the executable name.

Reply:
It probably doesn't matter as long you don't start multiple programs that are using the same UID3.

This is what I've found online regarding valid UID3 ranges:

- `0x10000000 – 0x1FFFFFFF`: Assigned by Symbian Ltd. / Nokia. Use only if officially registered.
- `0x20000000 – 0x2FFFFFFF`: Safe to use for internal/dev/test apps. No registration needed.
- `0xE0000000 – 0xEFFFFFFF`: Often used during development. Should not be shipped.
- `0x00000000 – 0x0FFFFFFF`: Avoid using. Many are taken by core OS components.

Symbian Ltd is defunct and you can't register UIDs nowadays, but I guess, everything within the second range should be fine.

Reply:
Thanks.

The tests/examples simply derive a uid3 of the md5 hash of its target name.
That's not really compliant with the "common Symbian practice".

I'd say this is a don't car, but feel free to apply a better fix.
(I'd like to avoid giving every test an explicit UID3)

Reply:
Fine with me. :)